cmd/prometheus/main.go

```diff
@@ -92,15 +94,17 @@ func Main() int {
 	}
 
 	var (
-		notifier      = notifier.New(&cfg.notifier)
-		targetManager = retrieval.NewTargetManager(sampleAppender)
-		queryEngine   = promql.NewEngine(memStorage, &cfg.queryEngine)
+		notifier                = notifier.New(&cfg.notifier)
+		targetManager           = retrieval.NewTargetManager(sampleAppender)
+		queryEngine             = promql.NewEngine(memStorage, &cfg.queryEngine)
+		queryCtx, cancelQueries = context.WithCancel(context.Background())
```

Comment:
Right... The maybe dumb reason why I was careful to call it a special query context along with a query cancellation function, was a vague worry about the shutdown order in https://github.com/prometheus/prometheus/blob/49d8f967746422877a78771ff8184f1b7f55b0eb/cmd/prometheus/main.go#L194-L196. As in, I didn't want that context to be used for other things than queries in the future, so that that shutdown order wouldn't be jeopardized. But I don't see a real problem with it. Even if other things would use and be cancelled by the same context in the future, the parallel nature of all that should work itself out fine.

Reading the code again, I would suggest getting rid of the `queryCtx` / `QueryCtx` naming everywhere and just making it a generic context that others are free to use as well (and use for being cancelled). What do you think?